js/README.md

```diff
@@ -100,55 +100,163 @@ Subdirectories with additional utilities
 This section describes how to add new blocks to Music Blocks in order
 to add functionality.
 
-* You can add an individual block to a palette by modifying two files:
-`basicblocks.js` and `logo.js`.
+<!-- * You can add an individual block to a palette by modifying two files:
+`basicblocks.js` and `logo.js`. -->
+* To add a new block you first need to look through `js/blocks/` and determine if a new file is needed or not. If required then go through the section down below on how to declare new files for blocks requiring so.
 
-* If the block you are adding needs to expand into a stack of blocks,
-you may also need to modify `macro.js`.
+## How to define new files for blocks
 
-* If you want to add a new palette with multiple blocks for a specific
+Note: You may skip this section if the block you're trying to add doesn't require a new file.
+
+* Make a new file in `js/blocks` with a meaningful name.
+
+e.g. Current files are named as  `GraphicsBlocks.js` , `MediaBlocks.js`.
+
+* Add that file to `MUSICBLOCKS_EXTRAS` in `js/activity.js`.
+
+* Create a setup function in your new file at the end, with a meaningful name.
+
+e.g. `setupGraphicsBlocks()`.
+
+* Call that setup function in `js/basicblocks.js` inside `initBasicProtoBlocks()` function.
+
+<!-- * If the block you are adding needs to expand into a stack of blocks,
+you may also need to modify `macro.js`. -->
+
+<!-- * If you want to add a new palette with multiple blocks for a specific
 application, you may want to write a plugin. Please see:
 [plugin](http://github.com/sugarlabs/musicblocks/tree/master/plugins)
-instead.
+instead. -->
 
-## How to define a new block in basicblocks.js (deprecated)
+## How to define a new block
 
+Note: You should directly start with this step if the block you want to add doesn't require a new file and makes use of an existing file inside `js/blocks`.
```

Comment:
Maybe something along the lines of find the file associated with the palette you want to add your new block to...

Reply:
Done. Please take a look.